src/PerfCounters.cc

```diff
@@ -34,10 +34,13 @@ namespace rr {
 static bool attributes_initialized;
 // At some point we might support multiple kinds of ticks for the same CPU arch.
 // At that point this will need to become more complicated.
-static struct perf_event_attr ticks_attr;
-static struct perf_event_attr minus_ticks_attr;
-static struct perf_event_attr cycles_attr;
-static struct perf_event_attr llsc_fail_attr;
+struct perf_event_attrs {
+  perf_event_attr ticks{};
+  perf_event_attr minus_ticks{};
+  perf_event_attr cycles{};
+  perf_event_attr llsc_fail{};
+};
+static std::vector<perf_event_attrs> perf_attrs;
```

Comment:
Please document what this vector is indexed by.

Reply:
What do you mean? The index here isn't actually very meaningful. This is more of a set than an ordered array I guess. I use vector since it's much simpler and it's index should be consistent with the fds.